girder/models/model_base.py

```diff
@@ -736,7 +750,96 @@ def setPublic(self, doc, public, save=False):
 
         return doc
 
-    def setAccessList(self, doc, access, save=False):
+    def setPublicFlags(self, doc, flags, user, append=False, save=False, force=False):
+        """
+        Set permission flags that are granted on this resource to anonymous users.
+        This means any user, whether anonymous or logged in, will receive all
+        of the specified permissions. This also validates that the user attempting
+        to set the flags has permission to do so.
+
+        :param doc: The document to update permission flags on.
+        :type doc: dict
+        :param flags: Flags or set of flags to add.
+        :type flags: flag identifier, or a list/set/tuple of them
+        :param user: The user performing this action.
+        :type user:
+        :param append: Whether to append to the list or replace it.
+        :type append: bool
+        :param save: Whether to save the document to the database afterward.
+        :type save: bool
+        :param force: Set this to True to disable validation of the current
+            user's ability to set the flags.
+        :type force: bool
+        """
+        currentFlags = doc.get('publicFlags', [])
+
+        if not isinstance(flags, (list, tuple, set)):
+            flags = [flags]
+
+        if append:
+            flags = currentFlags + list(flags)
+
+        flags = set(flags) & set(PERMISSION_FLAGS.keys())
+
+        if force or user['admin']:
+            doc['publicFlags'] = list(flags)
+        else:
+            allowedFlags = []
+            for flag in flags:
+                info = PERMISSION_FLAGS[flag]
+
+                # If this is an admin-only flag, we only allow it if it's already enabled.
+                if not info['admin'] or (info['admin'] and flag in currentFlags):
+                    allowedFlags.append(flag)
+
+            doc['publicFlags'] = allowedFlags
+
+        if save:
+            doc = self.save(doc)
+
+        return doc
+
+    def _isFlagEnabled(self, doc, type, entry, flag):
+        """
+        Check whether a specific flag is enabled for a specific ACL entry in the document.
+        """
+        id = entry['id']
+        perms = doc.get('access', {})
+
+        if type == 'group':
+            return self._hasGroupAccessFlag(perms.get('groups', ()), id, flag)
+        else:  # 'user'
```

Comment:
I find this comment surprising, for some reason I wouldn't have expected it from you here.  No action required.

😮